### ISSUE TITLE
HDF5 Import Export Simple

### ISSUE THREAD

**Issue by FLO-2DKaren** [FLO-2DSoftware/qgis-flo-2d-plugin#895]
Export data as hdf5 instead of *.dat.

- [x] cont
- [x] toler
- [x] neighbors
- [x] topo
- [x] mannings_n

**Comment by ldebek** [comment]
@FLO-2DKaren Neighbors are not used within plugin, so import/export is skipped for that component.

**Comment by ldebek** [comment]
PR: https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/pull/916

**Comment by ldebek** [comment]
@FLO-2DKaren In that case it will need some extra work as this data is not currently stored in a GeoPackage, so we need to add a new table if we want to support import + it would be good to have a look at exact algorithm used in FLO-2D engine. I could compare output with the one written in NEIGHBORS.DAT, but these one seems to be a binary format and I can't open it in a text editor as other DAT files.

**Comment by FLO-2DNoemi** [comment]
@Łukasz Dębek ***@***.***>  NEIGHBORS.DAT file
contains the neighboring cells in 8 directions, see below how FLO-2D code
writes the file:

WRITE(1709)(IFPNQ(I,J),J=1,8)

I goes from 1 to the number of cells in the domain.
J are the 8 directions, the four compass directions (1:north, 2:east,
3:south and 4:west) and the four diagonal directions (5:northeast,
6:southeast, 7:southwest and 8:northwest) ,  see image below.

In other words, the structure of the file is as follows:

IFPNQ(cell=1, direction=1), IFPNQ( cell=1, direction=2), IFPNQ(
cell=1,direction=3), IFPNQ( cell=1,direction=4), IFPNQ(
cell=1,direction=5), IFPNQ( cell=1,direction=6), IFPNQ(
cell=1,direction=7), IFPNQ( cell=1,direction=8)

[image: image.png]

I hope this helps. Please let me know if you have any questions.

Noemi

**Comment by FLO-2DKaren** [comment]
[FLO-2D.zip](https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/files/11251142/FLO-2D.zip)

**Comment by ldebek** [comment]
@FLO-2DKaren Calculating and exporting neighbors data into HDF5 format is available withing PR: https://github.com/FLO-2DSoftware/qgis-flo-2d-plugin/pull/922 Feel free to give it a try.
